### ISSUE TITLE
sso not working on gke w/ coredns

### ISSUE THREAD

**Issue by rwong2888** [argoproj/argo-cd#6839]
I have ha argocd deployed on gke, where i've scaled down kubedns to 0 and installed coredns.

I set up SSO w/ okta and dex following https://argoproj.github.io/argo-cd/operator-manual/user-management/okta/#saml-with-dex

However when I try to login via SSO, I am getting below error. It is trying to resolve argocd-redis via kubedns and not coredns. Any suggestions? Is there a way to use the coredns service?

```
failed to verify app state usLapOfhFz: dial tcp: lookup argocd-redis on 10.58.0.10:53: no such host
```

**Comment by alexmt** [comment]
Hello @rwong2888 , you might be facing https://github.com/argoproj/argo-cd/issues/6336 . Apparently redis go client caches IP address and does handle well redis service changes. Can you try to restart `argocd-server` and `argocd-application-controller` pods?

**Comment by rwong2888** [comment]
Issue still persists after restart @alexmt

**Comment by rwong2888** [comment]
Works when I remove HA install and redis goes to 1 pod and 0 haproxy.

Quoting text from slack conversation

@alexmt 
> it is day zero bug - apparently redis client does not handle IP changes well 
https://github.com/go-redis/redis/issues/1127
I plan to fix issue soon and cherry pick it into 2.0/2.1

**Comment by rwong2888** [comment]
Was able to fix with @alexmt suggestion below

> I think you should be good with HA if you add - --redis - "argocd-redis-ha-haproxy:6379" to your patch:
command:
```
- argocd-server
- --staticassets
- /shared/app
- --insecure
- --redis
- "argocd-redis-ha-haproxy:6379"
```
